include/bluetooth/mesh/light_ctl_srv.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_light_ctl_srv Light CTL Server model
+ * @{
+ * @brief API for the Light CTL Server model.
+ */
+
+#ifndef BT_MESH_LIGHT_CTL_SRV_H__
+#define BT_MESH_LIGHT_CTL_SRV_H__
+
+#include <bluetooth/mesh/light_ctl.h>
+#include <bluetooth/mesh/light_ctl_temp_srv.h>
+#include <bluetooth/mesh/model_types.h>
+#include <bluetooth/mesh/lightness_srv.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct bt_mesh_light_ctl_srv;
+
+/** @def BT_MESH_LIGHT_CTL_SRV_INIT
+ *
+ * @brief Initialization parameters for a @ref bt_mesh_light_ctl_srv instance.
+ *
+ * @param[in] _handlers Light CTL server callbacks.
+ */
+#define BT_MESH_LIGHT_CTL_SRV_INIT(_handlers)                                  \
+	{                                                                      \
+		.handlers = _handlers,                                         \
+		.lightness_srv = BT_MESH_LIGHTNESS_SRV_INIT(                   \
+			&_bt_mesh_light_ctl_lightness_handlers),               \
+		.temp_srv = BT_MESH_LIGHT_CTL_TEMP_SRV_INIT(                   \
+			&_bt_mesh_light_ctl_temp_handlers),                    \
+		.pub = { .msg = NET_BUF_SIMPLE(BT_MESH_MODEL_BUF_LEN(          \
+				 BT_MESH_LIGHT_CTL_STATUS,                     \
+				 BT_MESH_LIGHT_CTL_MSG_MAXLEN_STATUS)) },      \
+	}
+
+/** @def BT_MESH_MODEL_LIGHT_CTL_SRV
+ *
+ * @brief Light CTL Server model composition data entry.
+ *
+ * @param[in] _srv Pointer to a @ref bt_mesh_light_ctl_srv instance.
+ */
+#define BT_MESH_MODEL_LIGHT_CTL_SRV(_srv)                                      \
+	BT_MESH_MODEL_LIGHTNESS_SRV(&(_srv)->lightness_srv),                   \
+		BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_LIGHT_CTL_SRV,               \
+				 _bt_mesh_light_ctl_srv_op, &(_srv)->pub,      \
+				 BT_MESH_MODEL_USER_DATA(                      \
+					 struct bt_mesh_light_ctl_srv, _srv),  \
+				 &_bt_mesh_light_ctl_srv_cb),                  \
+		BT_MESH_MODEL_CB(BT_MESH_MODEL_ID_LIGHT_CTL_SETUP_SRV,         \
+				 _bt_mesh_light_ctl_setup_srv_op,              \
+				 &(_srv)->setup_pub,                           \
+				 BT_MESH_MODEL_USER_DATA(                      \
+					 struct bt_mesh_light_ctl_srv, _srv),  \
+				 NULL)
+
+/** Generic status response structure for CTL messages */
+struct bt_mesh_light_ctl_gen_cb_status {
+	struct bt_mesh_light_ctl_params current;
+	struct bt_mesh_light_ctl_params target;
+	int32_t remaining_time;
+};
+
+/** Generic set structure for CTL messages */
+struct bt_mesh_light_ctl_gen_cb_set {
+	uint16_t *light;
+	uint16_t *temp;
+	int16_t *delta_uv;
+	const struct bt_mesh_model_transition *transition;
+};
```

Comment:
Just pass these as raw parameters to the set call, perhaps? It breaks the conventional 4-parameter max limit, but it's not actually any slower.